src/Umbraco.Web.UI.Client/src/common/services/user.service.js

```diff
@@ -279,41 +301,32 @@ angular.module('umbraco.services')
             /** Loads the Moment.js Locale for the current user. */
             loadMomentLocaleForCurrentUser: function () {
 
-                function loadLocales(currentUser, supportedLocales) {
-                    var locale = currentUser.locale.toLowerCase();
-                    if (locale !== 'en-us') {
-                        var localeUrls = [];
-                        if (supportedLocales.indexOf(locale + '.js') > -1) {
-                            localeUrls.push('lib/moment/' + locale + '.js');
-                        }
-                        if (locale.indexOf('-') > -1) {
-                            var majorLocale = locale.split('-')[0] + '.js';
-                            if (supportedLocales.indexOf(majorLocale) > -1) {
-                                localeUrls.push('lib/moment/' + majorLocale);
-                            }
-                        }
-                        return assetsService.load(localeUrls, $rootScope);
-                    }
-                    else {
-                        //return a noop promise
-                        var deferred = $q.defer();
-                        var promise = deferred.promise;
-                        deferred.resolve(true);
-                        return promise;
-                    }
-                }
-
                 var promises = {
                     currentUser: this.getCurrentUser(),
                     supportedLocales: javascriptLibraryService.getSupportedLocalesForMoment()
                 }
 
                 return $q.all(promises).then(function (values) {
-                    return loadLocales(values.currentUser, values.supportedLocales);
+                    return services.loadLocales(values.currentUser.locale, values.supportedLocales);
                 });
+
+            },
+
+            /** Loads specific Moment.js Locales. */
+            loadLocales: function (locales, supportedLocales) {
                 
-                
+                var localeUrls = getMomentLocales(locales, supportedLocales);
 
+                if (localeUrls.length >= 1) {
+                    return assetsService.load(localeUrls, $rootScope);
+                }
+                else {
+                    //return a noop promise
+                    var deferred = $q.defer();

```

Comment:
a noop promise result is just : `return $q.when(true)` you don't need to create a promise and resolve it this way.

Reply:
@Shazwazza okay, but the original code already did it that way, so I didn't change that part: https://github.com/umbraco/Umbraco-CMS/pull/2984/files/e9f8fb8d1fd3f88ee921047edcfdf064bfc53dcb#diff-8f01a52bbe77f96aeb7228b59a06cca6